src/lib/drawer/DrawerProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useContext, useReducer } from 'react';
+import { drawerActions, DrawerActions, DrawerReducer } from './Drawer';
+
+const DrawerStateContext: React.Context<any> = React.createContext({});
+
+const DrawerActionContext: React.Context<DrawerActions> = React.createContext({} as DrawerActions);
+
+export const useDrawer = () => {
+  const drawerState = useContext(DrawerStateContext);
+  const drawerActions = useContext(DrawerActionContext);
```

Comment:
Why do we need two contexts for this. Could we not achieve what we need with one?